Tests/eppo/ClientTests.swift

```diff
@@ -108,307 +108,154 @@ final class eppoClientTests: XCTestCase {
-                    assignmentLogger: loggerSpy.logger,
-                    assignmentCache: nil)
-
+        eppoClient = EppoClient.configure(apiKey: "mock-api-key",
+                                assignmentLogger: loggerSpy.logger,
+                                assignmentCache: nil)
+        
         stub(condition: isHost("fscdn.eppo.cloud")) { _ in
             let stubData = RacTestJSON.data(using: .utf8)!
             return HTTPStubsResponse(data: stubData, statusCode: 200, headers: ["Content-Type": "application/json"])
         }
-        try await eppoClient.load()
+        
+        try await eppoClient.loadIfNeeded()
         
         _ = try eppoClient.getStringAssignment("6255e1a72a84e984aed55668", "randomization_algo")
         _ = try eppoClient.getStringAssignment("6255e1a72a84e984aed55668", "randomization_algo")
-
+        
         XCTAssertEqual(loggerSpy.logCount, 2, "Should log twice since there is no cache.")
     }
-
+    
     func testDoesNotLogDuplicateAssignmentsWithCache() async throws {
+        eppoClient = EppoClient.configure(
+            apiKey: "mock-api-key",
+            assignmentLogger: loggerSpy.logger,
+            // InMemoryAssignmentCache is default enabled.
+            assignmentCache: InMemoryAssignmentCache()
+        )
+        
         stub(condition: isHost("fscdn.eppo.cloud")) { _ in
             let stubData = RacTestJSON.data(using: .utf8)!
             return HTTPStubsResponse(data: stubData, statusCode: 200, headers: ["Content-Type": "application/json"])
         }
-        try await eppoClient.load()
+        
+        try await eppoClient.loadIfNeeded()
         
         _ = try eppoClient.getStringAssignment("6255e1a72a84e984aed55668", "randomization_algo")
         _ = try eppoClient.getStringAssignment("6255e1a72a84e984aed55668", "randomization_algo")
-
+        
         XCTAssertEqual(loggerSpy.logCount, 1, "Should log once due to cache hit.")
     }
     
     func testLogsForEachUniqueFlag() async throws {
+        eppoClient = EppoClient.configure(
+            apiKey: "mock-api-key",
+            assignmentLogger: loggerSpy.logger,
+            // InMemoryAssignmentCache is default enabled.
+            assignmentCache: InMemoryAssignmentCache()
+        )
+        
         stub(condition: isHost("fscdn.eppo.cloud")) { _ in
             let stubData = RacTestJSON.data(using: .utf8)!
             return HTTPStubsResponse(data: stubData, statusCode: 200, headers: ["Content-Type": "application/json"])
         }
-        try await eppoClient.load()
+        try await eppoClient.loadIfNeeded()
         
         _ = try eppoClient.getStringAssignment("6255e1a72a84e984aed55668", "randomization_algo")
         _ = try eppoClient.getStringAssignment("6255e1a72a84e984aed55668", "new_user_onboarding")
-
-        XCTAssertEqual(loggerSpy.logCount, 2, "Should log 2 times due to changing flags.")
-    }
-    
-    func testLoggingWhenRolloutIncreases() async throws {
```

Comment:
It is not possible to execute the network request multiple times so this test does not make sense anymore.